hibernate-core/src/main/java/org/hibernate/cache/spi/QueryKey.java

```diff
@@ -26,7 +26,7 @@ public class QueryKey implements Serializable {
 	/**
 	 * todo (6.0) : integrate work from original 6.0 branch
 	 */
-	public interface ParameterBindingsMemento {
+	public interface ParameterBindingsMemento extends Serializable {
```

Comment:
Is it possible similar issue still exists if the following `values` array contains object not implementing `Serializable`? Hopefully not.
```
	private static class ParameterBindingsMementoImpl implements QueryKey.ParameterBindingsMemento {
		private final Object[] values;
```

Reply:
Of course this can happen, but there is no way to magically make parameter values serializable which are provided by a user.

Reply:
yeah, makes sense.

Reply:
My original concern is whether we can refactor the `values` type to `Serializable[]`, but it might be complex and has ripple effect. The current approach is good enough.